### DISCUSSION TITLE
Tailwind not working with vue

### DISCUSSION THREAD

**Discussion by whytepeter** [tailwindlabs/tailwindcss#1666] [General]
i installed tailwindcss into a vuejs SPA did all the setup 
create a assets/css/tailwind.css and added the necessary base styles
imported it in the main.js file
create a postcss.config.js file and copied the required configuration from the official documentation

but the tailwind styles don't apply to my markups
inside the tailwind.css
`@tailwind base;

@tailwind components;

@tailwind utilities;
`

inside the postcss.config.js
`module.exports = {
  plugins: [
    // ...
    require("tailwindcss"),
    require("autoprefixer"),
    // ...
  ],
}
`

inside the main.js file
`import Vue from "vue"
import App from "./App.vue"
import "./registerServiceWorker"
import router from "./router"
import store from "./store"
import axios from "axios"
import "./assets/css/tailwind.css"
import firebase from "firebase/app"
import "firebase/firestore"
import "firebase/auth"`

The package.json file
`"dependencies": {
    "autoprefixer": "^9.7.6",
    "axios": "^0.19.2",
    "core-js": "^3.6.4",
    "firebase": "^7.14.2",
    "register-service-worker": "^1.7.1",
    "tailwindcss": "^1.4.0",
    "vue": "^2.6.11",
    "vue-router": "^3.1.6",
    "vuex": "^3.1.3"
  },`
I don't know what am doing wrong

**Comment by adamwathan** [comment]
I'm not sure if you can import your styles into `main.js`, usually you do it in `App.vue`.

**Reply by whytepeter** [reply]
before the upgrade...it works normally
after i created a new project and installed the new tailwind version 1.4
I think its cause I didn't include the purgecss in the tailwind.config file

**Reply by whytepeter** [reply]
what happens if dont add this in the tailwind.config file
`purge: [
    './src/**/*.html',
    './src/**/*.vue',
    './src/**/*.jsx',
  ],`

**Reply by whytepeter** [reply]
is there a whole new process i have to follow to install the new version in a vue project'

**Reply by ali5alkaf5** [reply]
> what happens if dont add this in the tailwind.config file `purge: [ './src/**/*.html', './src/**/*.vue', './src/**/*.jsx', ],`

This is the only correct answer that solved my problem! Thank you

**Comment by chrisjbarr** [comment]
Is it possible purge-css is purging your `@tailwind base` and `@tailwind components`? I had a similar issue (< 1.4 and `vue-cli-plugin-tailwind/purgecss`) and wrapped those two lines in my `tailwind.css` with 
```
/* purgecss start ignore */
@tailwind base;

@tailwind components;
/* purgecss end ignore */
```

Probably not the right solution - I'm planning on revisiting after I upgrade to 1.4 today and remove `vue-cli-plugin-tailwind/purgecss`. Add those and just debug a bit and see if that is the real issue, then go from there? Also, fwiw, I have my import in `main.js` not `App.vue`.

**Reply by chrisjbarr** [reply]
@whytepeter  - I just upgraded to 1.4 and removed `vue-cli-plugin-tailwind/purgecss` and had no problems.. just followed instructions on `tailwindcss.com` ... do you have public repo that reproduces your problem you can share we can take a peek to try and help?

**Comment by noook** [comment]
I am having the same problem here.
Steps:
+ Upgrade `tailwindcss` from `1.2` to `1.4.6`
+ Move config from `postcss.config.js` to `tailwind.config.js` like this:
```js
// tailwind.config.js
module.exports = {
  purge: {
    content: ['./public/**/*.html', './src/**/*.vue'],
    options: {
      whitelistPatterns: [
        /-(leave|enter|appear)(|-(to|from|active))$/,
        /^(?!(|.*?:)cursor-move).+-move$/,
        /^router-link(|-exact)-active$/,
        /tooltip/,
        /popover/,
        /notification/,
      ],
    }
  },
  // ...
}
```
+ Remove `postcss.config.js`

Now all my style is removed, there's litterally nothing applied.

I am sure the file is imported, as when I write `html { background-color: red; }` in the file where I import tailwind works perfectly.

I am using Vue in dev mode, with their development server.

I unfortunately cannot share the repository, I'll try to create one though.

**Reply by noook** [reply]
Ok nevermind, what happened was configuration conflict. At first, I didn't delete the `postcss.config.js` file, and there was still remaining purge settings. I reduced the config file to only this:
```js
const tailwindcss = require('tailwindcss');

module.exports = {
  plugins: [
    tailwindcss,
  ],
};
```

In summary:
+ `postcss.config.js` must not be removed
+ Don't duplicate configuration

And it does the trick 😄

**Reply by SamuelALSN** [reply]
I did the same configuration but it tailwind still not working in my Vue-cli project

**Reply by nbpalomino** [reply]
I was having the same problem, but @adamwathan points to [this correct config for App.vue using tailwindcss](https://github.com/tailwindlabs/designing-with-tailwindcss/blob/master/04-building-a-dropdown-menu/01-styling-the-basic-dropdown-elements/src/App.vue)

And my problem was solved!

**Reply by mr-feek** [reply]
@nbpalomino I wish this was in the docs -- that got it working for me!

**Reply by PabloGnesutta** [reply]
YESSSSSSSSSSSSSSSSSSSSSSSSS! This is the one.

**Comment by ahmad721721** [comment]
just run these two commands and your are good to go
npm uninstall tailwindcss postcss autoprefixer
npm install tailwindcss@npm:@tailwindcss/postcss7-compat postcss@^7 autoprefixer@^9

postcss 8 is new so rolling back to postcss 7 for a while. once the issue with postcss 8 is resolved by tailwind dev team then you can upgrade to postcss by these commands

npm uninstall tailwindcss @tailwindcss/postcss7-compat
npm install tailwindcss@latest postcss@latest autoprefixer@latest

**Reply by black** [reply]
This worked for me ....awesome :) : )

**Reply by AniSenpai10** [reply]
Worked like a charm, shot bud.

**Reply by jkirira** [reply]
This worked. Thanks for this.

**Reply by tonyIkwuegbu** [reply]
Thanks Ahmad

**Reply by Bahrul-Rozak** [reply]
This worked, thanks man

**Reply by Monoxt** [reply]
Thanks Ahmad

**Reply by kakcruz** [reply]
Thanks!!!! Save my life!!

**Comment by kylewill** [comment]
I've got the same issue using this starter template: https://github.com/kefranabg/bento-starter

I've tried all of the steps in this thread including the postcss 7 rollback, using in App.vue vs. main.js. No TailwindCSS is rendered in the project.

If I start a new project with Vue 3 + Vite using the docs it works. But I'm already down the road on this project using the bento starter.

edit: is there any debug log and I share to determine where the breakdown is?

**Reply by itrew** [reply]
Have you tried just [disabling the purgecss piece](https://github.com/forsartis/vue-cli-plugin-tailwind#configure-purgecss) of tailwind to see if that solves your issue? I was running into an issue with writing custom styles for inner elements of [Vue Formulate](https://vueformulate.com/) so some of my styles were actually defined in `main.js` and were getting dropped via `purgecss`. Ended up adding a path to `purge.content` that now includes all the classes NOT defined in my `.vue` files.

```javascript
// tailwind.config.js
module.exports = {
  purge: {
    enabled: true,
    content: ['./public/**/*.html', './src/**/*.vue', './src/main.js'],
    // ...
  },
  // ...
}
```

**Comment by hussain5577** [comment]
### Replace this code with your postcss.config.js

```js
module.exports = () => {
  return {
    postcssPlugin: 'postcss-will-change',
    Declaration: {
      'will-change': (decl, { Declaration }) => {
        decl.cloneBefore(
          new Declaration({ prop: 'backface-visibility', value: 'hidden' })
        )
      }
    }
  }
}
module.exports.postcss = true
```
[https://github.com/postcss/postcss/releases/tag/8.0.0](url)

**Comment by jenstornell** [comment]
The docs are quite solid for installing Vue + Tailwind.

https://tailwindcss.com/docs/guides/vue-3-vite

**Reply by tony-nyagah** [reply]
This setup doesn't work if you're not using Vite from my experience.

**Reply by snakysnake** [reply]
I did this 1:1 with Vite and Vue 3 and still cant get it to work

**Reply by WallaceSociety** [reply]
The instructions fail to mention that you need to add tailwindcss: {}, to your plugins object in postcss.config.js 

```
module.exports = {
    plugins: {
      tailwindcss: {},
      autoprefixer: {}
    }
  }
```

**Reply by alisalehi1380** [reply]
Tnks dude! work for me. 👍

![image](https://github.com/tailwindlabs/tailwindcss/assets/111766206/e3e8ca49-09d0-44cd-9dbb-ae235720343a)

**Comment by Jehadalmaliki** [comment]
The most issue programmer faced I think they didn't read carefully documentation and search about the specific using of tailwind with witch framework does it apply ,the will save there time when they apply the tips mentions on .

**Reply by blue6888** [reply]
please you download tailwind in your vue , you can try it if ok

**Comment by KarmaBlackshaw** [comment]
I tried solutions above to no success. I followed [this one](https://morioh.com/p/b1c3d288fcde). 

Technically, you have to create `postcss.config.js`
```js
module.exports = {
    plugins: [
        require('tailwindcss'),
        require('autoprefixer')
    ]
}
```

```js
"vue": "^2.7.14",
"autoprefixer": "^10.4.13",
"postcss": "^8.4.21",
"tailwindcss": "^3.2.4",
```

**Comment by tonyIkwuegbu** [comment]
Thanks

**Comment by mrrajsoni** [comment]
https://tailwindcss.com/docs/guides/vite#vue - Following this solved my issue.

I was missing .{vue} from the tailwind.config.js content property

` content: ['./index.html', './src/**/*.{vue,js,ts,jsx,tsx}'],`

Adding .vue extension solved it.

**Reply by Afsalazeez** [reply]
Thanks a lot! It worked !

**Reply by LuckyArdhika** [reply]
work like a charm!, thats not explained in starter docs

**Reply by hdwts** [reply]
Its work with **Vue 3.2.13 & tailwind 3.3.3**

With below config 

["dependencies": {
    "core-js": "^3.8.3",
    "vue": "^3.2.13"
  },
  "devDependencies": {
    "@babel/core": "^7.12.16",
    "@babel/eslint-parser": "^7.12.16",
    "@vue/cli-plugin-babel": "~5.0.0",
    "@vue/cli-plugin-eslint": "~5.0.0",
    "@vue/cli-service": "~5.0.0",
    "autoprefixer": "^10.4.15",
    "eslint": "^7.32.0",
    "eslint-plugin-vue": "^8.0.3",
    "postcss": "^8.4.29",
    "tailwindcss": "^3.3.3"
  }](url)

Thanks @mrrajsoni 👍

**Reply by alisalehi1380** [reply]
Tnks man 🙏
